java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java

```diff
@@ -1325,72 +1229,59 @@ public CaseGroupTreeImpl switchGroup(List<CaseLabelTreeImpl> labels, BlockStatem
     return new CaseGroupTreeImpl(labels, blockStatements);
   }
 
-  public CaseLabelTreeImpl newCaseSwitchLabel(AstNode caseToken, ExpressionTree expression, AstNode colonToken) {
-    InternalSyntaxToken caseSyntaxToken = InternalSyntaxToken.create(caseToken);
-    InternalSyntaxToken colonSyntaxToken = InternalSyntaxToken.create(colonToken);
-    return new CaseLabelTreeImpl(caseSyntaxToken, expression, colonSyntaxToken);
+  public CaseLabelTreeImpl newCaseSwitchLabel(JavaTree caseSyntaxToken, ExpressionTree expression, InternalSyntaxToken colonSyntaxToken) {
+    return new CaseLabelTreeImpl((InternalSyntaxToken) caseSyntaxToken, expression, colonSyntaxToken);
   }
 
-  public CaseLabelTreeImpl newDefaultSwitchLabel(AstNode defaultToken, AstNode colonToken) {
-    InternalSyntaxToken defaultSyntaxToken = InternalSyntaxToken.create(defaultToken);
-    InternalSyntaxToken colonSyntaxToken = InternalSyntaxToken.create(colonToken);
-    return new CaseLabelTreeImpl(defaultSyntaxToken, null, colonSyntaxToken);
+  public CaseLabelTreeImpl newDefaultSwitchLabel(JavaTree defaultToken, InternalSyntaxToken colonToken) {
+    InternalSyntaxToken defaultSyntaxToken = (InternalSyntaxToken) defaultToken;
+    return new CaseLabelTreeImpl(defaultSyntaxToken, null, colonToken);
   }
 
-  public SynchronizedStatementTreeImpl synchronizedStatement(AstNode synchronizedToken, AstNode openParen, ExpressionTree expression, AstNode closeParen, BlockTreeImpl block) {
-    InternalSyntaxToken synchronizedKeyword = InternalSyntaxToken.create(synchronizedToken);
-    InternalSyntaxToken openParenToken = InternalSyntaxToken.create(openParen);
-    InternalSyntaxToken closeParenToken = InternalSyntaxToken.create(closeParen);
+  public SynchronizedStatementTreeImpl synchronizedStatement(JavaTree synchronizedToken, InternalSyntaxToken openParenToken, ExpressionTree expression, InternalSyntaxToken closeParenToken, BlockTreeImpl block) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Split this 211 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00103)